### PR TITLE
Functional tests - Add test 'Sort Addresses'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add functional test `Sort Addresses`
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `TEST_PATH="functional/BO/04_customers/02_addresses/04_sortAddresses.js" URL_FO=shopUrl/ npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17772)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    await testContext.addContextItem(this, 'testIdentifier', 'goToAddressesPage', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.customersParentLink,
+      this.pageObjects.boBasePage.addressesLink,
+    );
+    const pageTitle = await this.pageObjects.addressesPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.addressesPage.pageTitle);
+  });
+
+  it('should reset all filters and get Number of customers in BO', async function () {
```
Comment (PierreRambaud) [inline tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js]:
```suggestion
  it('should reset all filters and get number of addresses in BO', async function () {
```

**Review by PierreRambaud — DISMISSED** [review]

File: tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  const tests = [
+    {
+      args: {
+        testIdentifier: 'sortByIdDesc', sortBy: 'id_address', sortDirection: 'desc', isFloat: true,
+      },
+    },
+    {args: {testIdentifier: 'sortByFirstNameAsc', sortBy: 'firstname', sortDirection: 'asc'}},
+    {args: {testIdentifier: 'sortByFirstNameDesc', sortBy: 'firstname', sortDirection: 'desc'}},
+    {args: {testIdentifier: 'sortByLastNameCodeAsc', sortBy: 'lastname', sortDirection: 'asc'}},
+    {args: {testIdentifier: 'sortLastNameDesc', sortBy: 'lastname', sortDirection: 'desc'}},
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js]:
```suggestion
    {args: {testIdentifier: 'sortByLastNameDesc', sortBy: 'lastname', sortDirection: 'desc'}},
```

File: tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    {args: {testIdentifier: 'sortByAddress1Asc', sortBy: 'address1', sortDirection: 'asc'}},
+    {args: {testIdentifier: 'sortAddress1Desc', sortBy: 'address1', sortDirection: 'desc'}},
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js]:
```suggestion
    {args: {testIdentifier: 'sortByAddress1Desc', sortBy: 'address1', sortDirection: 'desc'}},
```

File: tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    {args: {testIdentifier: 'sortByPostCode1Asc', sortBy: 'postcode', sortDirection: 'asc'}},
+    {args: {testIdentifier: 'sortPostCodeDesc', sortBy: 'postcode', sortDirection: 'desc'}},
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js]:
```suggestion
    {args: {testIdentifier: 'sortByPostCode1Desc', sortBy: 'postcode', sortDirection: 'desc'}},
```

File: tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    {args: {testIdentifier: 'sortByCityAsc', sortBy: 'city', sortDirection: 'asc'}},
+    {args: {testIdentifier: 'sortCityDesc', sortBy: 'city', sortDirection: 'desc'}},
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/04_customers/02_addresses/04_sortAddresses.js]:
```suggestion
    {args: {testIdentifier: 'sortByCityDesc', sortBy: 'city', sortDirection: 'desc'}},
```

**Review by SimonGrn — COMMENTED** [review]
Thanks @boubkerbribri

**Review by SimonGrn — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**